### PR TITLE
Staff of Necromancy Update: Rise of the Meatomancer

### PR DESCRIPTION
**Pull request by GreatBigFailure**
The staff of necromancy can now, in addition to all its previous functions, raise simple mobs, monkeys, and meat. 

**Mobs**
Mobs and monkeys will become a slower version of what they were in life. Any special traits like ranged attacks or abilities are not carried over, they're dumb zombies that smack the crew around. This also gibs the animal's old shell, conveniently making a slice of meat.

**Meat**
Necromancy is well and good but Meatomancy is the hip new thing all the wiz-kids are trying out. Using the staff of necromancy on a chunk of meat will create a "Meat Ghoul". The ghoul is a little quick and not too strong but easy to make. Their attack scales slightly with what meat they were and their health scales with how much nutriment the meat had. 
meatghoul

**Lich Secrets**
When in the possession of a Lich the staff of necromancy recharges twice as fast. It also heals the Lich of brute damage at a moderate rate.

mightylich

:cl:
 * rscadd: The Staff of Necromancy now has additional options for undead minions, become the meat-o-mancer!

### CODE REVIEW

**Comment by GreatBigFailure** [comment]
I'm curious about the thumbs down. This was a semi-quick goofy additions. What don't you like about it? I can make changes.

File: code/modules/projectiles/guns/energy/special.dm [hunk 1]
```diff
@@ -193,26 +198,64 @@
...
-		to_chat(user, "<span class = 'warning'>[!H.stat?"\The [target] needs to be dead or in a critical state first.":H.health>config.health_threshold_crit?"\The [target] has not received enough damage.":"Something went wrong with the conversion process."]</span>")
+	if(!charges || istype(target, /mob/living/simple_animal/hostile/necro) || get_dist(target, user) > 7)
 		return 0
+	var/toRaise = canRaise(target)
+	if(!toRaise)
+		return
+	switch(toRaise)
+		if(RAISE_HUMAN)
+			var/mob/living/carbon/human/H = target
+			if(!H.stat || (H.stat < DEAD && H.health > config.health_threshold_crit))
```
Comment (Zth--) [inline code/modules/projectiles/guns/energy/special.dm]: I know you didn't write this but this should be rewritten.

`if(H.stat != DEAD)` is probably the best

Comment (Zth--) [reply code/modules/projectiles/guns/energy/special.dm]: actually move all these checks to `canRaise()`

File: code/modules/mob/living/simple_animal/hostile/necro.dm [hunk 2]
```diff
@@ -33,6 +33,92 @@
...
+	move_to_delay = 3
+	maxHealth = 10
+	health = 10
+	melee_damage_lower = 5
+	melee_damage_upper = 8
+	attacktext = "meats"
+	environment_smash_flags = 0
+	var/bites = 3
+
+/mob/living/simple_animal/hostile/necro/meat_ghoul/proc/meatOfTheDead(M)
```
Comment (Zth--) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]: This proc name is very confusing, it should be more clear on what it does

File: code/modules/mob/living/simple_animal/hostile/necro.dm [hunk 3]
```diff
@@ -33,6 +33,92 @@
...
+		speed = ghoulToBe.speed*2	//Slower than we were
+		maxHealth = ghoulToBe.maxHealth
+		health = maxHealth
+		melee_damage_upper = ghoulToBe.melee_damage_upper
+		melee_damage_lower = ghoulToBe.melee_damage_lower
+		attacktext = ghoulToBe.attacktext
+		speak = ghoulToBe.speak
+	zombIcon.ColorTone("#85B060")
+	icon = zombIcon
+	name = "[aGhoul.name] " + "ghoul"
```
Comment (Zth--) [inline code/modules/mob/living/simple_animal/hostile/necro.dm]:
```suggestion
	name = "[aGhoul.name] ghoul"
```

**Pull request merged**